### PR TITLE
Add Model logits output ensemble

### PR DESCRIPTION
**Pull request by zhijiejia**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Convenient segmentation probability ensemble of multiple models

## Modification

Added `model_ensemble.py` files, added `aug_test_logits` functions, in `mmseg/models/segmentors/encoder_decoder.py`

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmseg/models/segmentors/encoder_decoder.py [hunk 1]
```diff
@@ -300,3 +300,23 @@ def aug_test(self, imgs, img_metas, rescale=True):
...
+        """
+        # aug_test rescale all imgs back to ori_shape for now
+        assert rescale
+        # to save memory, we get augmented seg logit inplace
+
+        imgs = img
+        seg_logit = self.inference(imgs[0], img_metas[0], rescale)
+        for i in range(1, len(imgs)):
+            cur_seg_logit = self.inference(imgs[i], img_metas[i], rescale)
+            seg_logit += cur_seg_logit
```
Comment (MeowZheng) [inline mmseg/models/segmentors/encoder_decoder.py:318]: I have a little question why not divide `seg_logit` by `len(imgs)` that is the number of augmented data

**Review by MeowZheng — COMMENTED** [review]
why not ensemble results of models inference without tta?

**Comment by MeowZheng** [comment]
Would you mind providing your email to us? It is convenient for us to contact offline

File: mmseg/models/segmentors/encoder_decoder.py (hunk 1 above)
Comment (zhijiejia) [reply mmseg/models/segmentors/encoder_decoder.py:318]: yes, the `seg_logit` should divid `len(imgs)`, i have a mistake, even it will have same result with `divid len(imgs)` by `torch.argmax`, but `divid len(imgs)` is better.

By the way, do you think this needs to be added to this repo? Or are you willing to accept this idea (add model_ensemble code to repo)?

Comment (MeowZheng) [reply mmseg/models/segmentors/encoder_decoder.py:318]: Would you mind providing your email to us? It is convenient for us to contact offline

Comment (zhijiejia) [reply mmseg/models/segmentors/encoder_decoder.py:318]: > Would you mind providing your email to us? It is convenient for us to contact offline



> commented

`22151338@zju.edu.cn`

**Comment by zhijiejia** [comment]
> why not ensemble results of models inference without tta?

Because I originally wanted to get better results in the competition, using the tta method would be higher than not using it, so I used tta by default.

File: mmseg/models/segmentors/encoder_decoder.py (hunk 1 above)
Comment (MeowZheng) [reply mmseg/models/segmentors/encoder_decoder.py:318]: Please check your email, I have sent our invitation.
